Условия задачи

Leilorr opened the pull request:
Реализовать возможность удаления данной ранее рекомендации. Рекомендацию один пользователь дает другому после совместной работы над чем-то.

Leilorr committed:
Условия задачи
Реализовать возможность удаления данной ранее рекомендации. Рекомендацию один пользователь дает другому после совместной работы над чем-то.

Leilorr committed:
Условия задачи
Реализовать возможность удаления данной ранее рекомендации. Рекомендацию один пользователь дает другому после совместной работы над чем-то.

Leilorr committed:
Delete src/main/java/school/faang/user_service/service/SubscriptionService.java

Leilorr committed:
Merge remote-tracking branch 'origin/feature-BJS2-1455' into feature-BJS2-1455

InnMish commented on src/main/java/school/faang/user_service/service/RecommendationService.java:
```diff
@@ -0,0 +1,14 @@
+@RequiredArgsConstructor
+public class RecommendationService {
+    private final RecommendationRepository recommendationRepository;
+    public void delete(long id) {
```
сделай тут отступ между полями и методами, чтобы явно разделение видеть

InnMish replied:
Обрати внимание, в контроллере у тебя Long (обертка), а здесь long - примитив 
Какая проблема? 
Тебе в контроллер от пользователя может прийти null. Мы не должны такого допустить, правильно же? Если пользователь кого-то хочет удалить, пусть указывает тогда кого именно. 

Нужно сделать так, чтобы мы не разрешали пользователю отправлять запрос с null  
для этого надо поменять тип входного параметра в контроллере

InnMish replied:
примитив как раз не может быть null

InnMish replied:
Еще было бы удобно, если бы мы пользователю вернули данные после удаления профиля
Типа, user 1 удален 
Чтобы было прям видно, что кого-то удалили 
То есть в методе надо сделать возвращаемое значение UserDto

InnMish commented on src/test/java/school/faang/user_service/service/controller/RecommendationControllerTest.java:
```diff
@@ -0,0 +1,34 @@
+        recommendationController.deleteRecommendation(id);
+
+        verify(recommendationService, times(1)).delete(id);
+    }
```
Тут ты тестируешь, что контроллер у тебя дернет метод сервиса 
А еще надо написать метод, который проверит, что сервис дернет метод репозитория 

Также обрати внимание на метод deleteById, которорый ты в сервисе вызваешь. Проверь, есть ли у этого метода возвращаемое значение? Если есть, надо проверить, что тебе именно оно вернется после удаления)

Leilorr committed:
Получить все рекомендации пользователя
Условия задачи
Реализовать возможность получения всех рекомендаций, которые были даны одному пользователю.

Leilorr committed:
Получить все данные рекомендации
Условия задачи
Реализовать возможность получения всех рекомендаций, которые пользователь дал другим пользователям.

InnMish committed:
fixes after wrong comment :)

InnMish committed:
Merge branch 'feature-BJS2-1401' into feature-BJS2-1455

# Conflicts:
#	src/main/java/school/faang/user_service/controller/RecommendationController.java
#	src/main/java/school/faang/user_service/dto/recommendation/RecommendationDto.java
#	src/main/java/school/faang/user_service/mapper/RecommendationMapper.java
#	src/main/java/school/faang/user_service/service/RecommendationService.java
#	src/test/java/school/faang/user_service/service/RecommendationServiceTest.java

InnMish committed:
fixed conflicts
